### ISSUE TITLE
completed-docs false positive for local/nested functions

### ISSUE THREAD

**Issue by 0815fox** [palantir/tslint#4615]
### Bug Report

- __TSLint version__: Introduced with 5.14.0
- __TypeScript version__: 3.3.4000
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
/**
 * Doc for outer
 */
function outer(): number {
    /**
     * Doc for inner
     */
    function inner(): number {
        return 42;
    }
    return inner();
}
```

with `tslint.json` configuration:

```json
...
completed-docs: true
...
```

#### Actual behavior
Error reported on function inner: *Documentation must exist for public methods.*

#### Expected behavior
No error reported

**Comment by JoshuaKGoldberg** [comment]
Interesting. Some folks might prefer to enforce doc comments on those inner methods as well. Maybe the rule's config could provide a way to opt out of enforcing these?

**Comment by 0815fox** [comment]
Might be true. But then the message is slightly wrong in that case, as it is not a public method (okay, both functions are not methods...)

**Comment by Planeshifter** [comment]
We have been encountering a related bug with a slightly different lint error message. The linter complains about the following code:

```ts
/**
* Returns an iterator protocol-compliant object.
*
* @returns iterator protocol-compliant object
*/
function iterator() {
	return {
		'next': next
	};

	/**
	* Implements the iterator protocol `next` method.
	*
	* @returns iterator protocol-compliant object
	*/
	function next() {
		return {
			'value': true,
			'done': false
		};
	}
}
```
Specifically, it complains about missing documentation for `next`:

``` 
ERROR: xx:2 completed-docs  Documentation must exist for functions.
``` 
where `xx` refers to the line starting with `function next()`.

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
